make/test.mk

```diff
@@ -101,25 +101,22 @@ ifeq ($(E2E_REPO_PATH),"")
 			$(eval AUTHOR_LINK = https://github.com/${AUTHOR})
         else
 			$(eval AUTHOR_LINK = $(shell jq -r '.refs[0].pulls[0].author_link' <<< $${CLONEREFS_OPTIONS} | tr -d '[:space:]'))
-			@echo "using pull sha ${PULL_PULL_SHA}"
-			# get branch ref of the fork the PR was created from
-			$(eval BRANCH_REF := $(shell curl ${AUTHOR_LINK}/host-operator.git/info/refs?service=git-upload-pack --output - /dev/null 2>&1 | grep -a ${PULL_PULL_SHA} | awk '{print $$2}'))
+			@echo "found author link ${AUTHOR_LINK}"
+			$(eval BRANCH_NAME := $(shell jq -r '.refs[0].pulls[0].head_ref' <<< $${CLONEREFS_OPTIONS} | tr -d '[:space:]'))
         endif
 		@echo "using author link ${AUTHOR_LINK}"
-		@echo "detected branch ref ${BRANCH_REF}"
+		@echo "detected branch ${BRANCH_NAME}"
 		# check if a branch with the same ref exists in the user's fork of toolchain-e2e repo
-		$(eval REMOTE_E2E_BRANCH := $(shell curl ${AUTHOR_LINK}/toolchain-e2e.git/info/refs?service=git-upload-pack --output - 2>/dev/null | grep -a "${BRANCH_REF}$$" | awk '{print $$2}'))
+		$(eval REMOTE_E2E_BRANCH := $(shell curl ${AUTHOR_LINK}/toolchain-e2e.git/info/refs?service=git-upload-pack --output - 2>/dev/null | grep -a "refs/heads/${BRANCH_NAME}$$" | awk '{print $$2}'))
```

Comment:
this could detect another branch name with the same suffix, but different prefix.
Imagine that I want to pair PRs with branch name `cool-test`, but in e2e repo I have also `my-cool-test`

Reply:
I'm idiot :man_facepalming: 
of course that the prefix does the job. I don't understand how I reviewed the line. 
apologies for the confusion :pray: